src/server/webpack-plugin/util.js

```diff
@@ -21,4 +21,21 @@ export const validate = compiler => {
   }
 }
 
+export const onEmit = (compiler, name, hook) => {
+  if (compiler.hooks) {
+    // Webpack >= 4.0.0
+    compiler.hooks.emit.tap(name,
```

Comment:
Use `tapPromise` when returning a Promise.
But in this case it makes more sense to use `tapAsync` which accepts a function with callback.

Reply:
Thanks for your review @sokra! :) Does the new changes look good to you?